core/modules/layout/layout.flexible.inc

```diff
@@ -0,0 +1,1030 @@
+  layout_set_flexible_tempstore($template_data, $form_state['flexible_template_id']);
+  layout_flexible_reset_caches();
+}
+
+/*
+ * Inserts a new key/value before/after a key in the array.
+ * Code modified from http://eosrei.net/comment/287
+ *
+ * @param $key
+ *   The key to insert.
+ * @param $array
+ *   An array to insert in to.
+ * @param $new_key
+ *   The key to insert.
+ * @param $new_value
+ *   An value to insert.
+ *
+ * @return
+ *   The new array if the key exists, FALSE otherwise.
+ */
+function layout_flexible_array_insert_key_value($key, array &$array, $new_key, $new_value, $position) {
+  if (array_key_exists($key, $array)) {
+    $new = array();
+    if ($position == 'before') {
+      foreach ($array as $k => $value) {
+        if ($k === $key) {
+          $new[$new_key] = $new_value;
+        }
+        $new[$k] = $value;
+      }
+    }
+    else {
+      foreach ($array as $k => $value) {
+        $new[$k] = $value;
+        if ($k === $key) {
+          $new[$new_key] = $new_value;
+        }
+      }
+    }
+    return $new;
+  }
+  return FALSE;
+}
+
+/**
+ * AJAX responder to add a new row, region or region to a flexible layout.
+ */
+function layout_flexible_template_edit_region_style_ajax($form, $form_state) {
+  return $form['row_settings'];
+}
+
+/**
+ * AJAX responder to add a new row, region or region to a flexible layout.
+ */
+function layout_flexible_template_edit_row_ajax($form, $form_state) {
+  $commands = array();
+  // Display error messages in the form if any.
+  if (form_get_errors()) {
+    $html = '';
+    $html .= theme('status_messages');
+    $html .= backdrop_render($form);
+    $title = isset($form['#title']) ? $form['#title'] : backdrop_get_title();
+    $commands[] = ajax_command_open_modal_dialog($title, $html, array('dialogClass' => 'layout-dialog'));
+  }
+  else {
+    $commands[] = ajax_command_close_modal_dialog();
+
+    $update_ajax = layout_ajax_form_update($form, $form_state);
+    $commands = array_merge($commands, $update_ajax['#commands']);
+  }
+
+  return array(
+    '#type' => 'ajax',
+    '#commands' => $commands,
+  );
+}
+
+
+/**
+ * Form to delete a row from a flexible template.
+ *
+ * @param $flexible_template_id
+ *   The template to which a row is being added.
+ * @param $original_row
+ *   The row above or below which a new row is being inserted.
+ *
+ * @ingroup forms
+ */
+function layout_flexible_template_delete_row($flexible_template_id, $original_row) {
+  // Todo use token, like in block add page.
+
+  $commands = array();
+  $template = layout_flexible_template_template($flexible_template_id);
+  unset($template['regions'][$original_row]);
+  layout_set_flexible_tempstore($template, $flexible_template_id);
+
+  if (backdrop_is_ajax()) {
+    $commands[] = ajax_command_remove('#flexible-row-id-' . $original_row);
+
+    return array(
+      '#type' => 'ajax',
+      '#commands' => $commands,
+    );
+  }
+  else {
```

Comment:
This can be above with logic `if (! backdrop_is_ajax())`.

Reply:
It would still require an else statement if the negation came first.

Reply:
Or even better just remove the else conditional. It has an early return which already serves the same purpose.

Reply:
> It has an early return

True. This is just a copy and paste though. This repeats a few times in `layout.admin`